### PR TITLE
feat: show "continue" screen after successful verification

### PR DESCRIPTION
**Pull request by jonas-jonas**
The `link` strategy for verification now shows a confirmation screen with a "continue" link after successful verification, aligning its behavior to the `code` strategy.

Also fixes a bug, where the `default_browser_return_url` of the verification flow was not respected when using the code strategy.

## Related issue(s)

Closes ory-corp/cloud#3925
Fixes ory/network#228
Fixes https://github.com/ory/network/issues/224

## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [ ] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got the approval (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](https://github.com/ory/docs).

## Further Comments

https://user-images.githubusercontent.com/5540255/218118857-9fdefc4e-afce-4798-86d6-409b073b2b4f.mov

(_The red color of the confirmation stems from an outdated version of the kratos selfservice example UI. This is fixed in current versions of the repo_).

### CODE REVIEW

File: selfservice/strategy/link/strategy_verification_test.go [hunk 1]
```diff
@@ -361,39 +360,57 @@ func TestVerification(t *testing.T) {
...
-		flow, token := newValidFlow(t, public.URL+verification.RouteInitBrowserFlow+"?"+url.Values{"return_to": {returnToURL}}.Encode())
+		for _, fType := range []flow.Type{flow.TypeBrowser, flow.TypeAPI} {
+			t.Run(fmt.Sprintf("type=%s", fType), func(t *testing.T) {
+				client := testhelpers.NewClientWithCookies(t)
+				flow, token := newValidFlow(t, fType, public.URL+verification.RouteInitBrowserFlow)
 
-		body := fmt.Sprintf(
-			`{"csrf_token":"%s","email":"%s"}`, flow.CSRFToken, verificationEmail,
-		)
+				res, err := client.Get(public.URL + verification.RouteSubmitFlow + "?" + url.Values{"flow": {flow.ID.String()}, "token": {token.Token}}.Encode())
```
Comment (jonas-jonas) [inline selfservice/strategy/link/strategy_verification_test.go:397]: `Get`ing the link matches more closely, what a user would do (e.g. opening the link in a browser).

File: selfservice/flow/verification/flow_test.go [hunk 2]
```diff
@@ -142,3 +143,69 @@ func TestFromOldFlow(t *testing.T) {
...
+	const flowReturnTo = "https://ory.sh/flow-return-to"
+
+	for _, tc := range []struct {
+		desc       string
+		prep       func(conf *config.Config)
+		requestURL string
+		expect     string
+	}{
+		{
+			desc: "return_to has precendence over global return to",
```
Comment (Benehiko) [inline selfservice/flow/verification/flow_test.go:159]:
```suggestion
			desc: "return_to has precedence over global return to",
```

File: selfservice/flow/verification/flow_test.go [hunk 3]
```diff
@@ -142,3 +143,69 @@ func TestFromOldFlow(t *testing.T) {
...
+			requestURL: fmt.Sprintf("http://kratos:4433/verification?return_to=%s", localReturnTo),
+			expect:     globalReturnTo,
+		},
+		{
+			desc:       "with invalid request url",
+			requestURL: string([]byte{0x7f}), // 0x7f is an ASCII control char, and fails URL validation
+			expect:     globalReturnTo,
+		},
+		{
+			desc: "flow return to has precendence over global return to",
```
Comment (Benehiko) [inline selfservice/flow/verification/flow_test.go:177]:
```suggestion
			desc: "flow return to has precedence over global return to",
```

File: selfservice/flow/verification/flow_test.go [hunk 4]
```diff
@@ -142,3 +143,69 @@ func TestFromOldFlow(t *testing.T) {
...
+			prep: func(conf *config.Config) {
+				conf.MustSet(context.Background(), config.ViperKeySelfServiceVerificationBrowserDefaultReturnTo, flowReturnTo)
+			},
+			requestURL: "http://kratos:4433/verification",
+			expect:     flowReturnTo,
+		},
+		{
+			desc: "return_to has precendence over flow return to",
```
Comment (Benehiko) [inline selfservice/flow/verification/flow_test.go:185]:
```suggestion
			desc: "return_to has precedence over flow return to",
```

**Review by Benehiko — CHANGES_REQUESTED** [review]
Small nits, but other than that this PR is good :)

**Review by Benehiko — APPROVED** [review]
Very nice! :)

**Review by aeneasr — APPROVED** [review]

**Pull request merged**